src/transformers/models/udop/tokenization_udop.py

```diff
@@ -0,0 +1,1520 @@
+            [8774, 32099, 3, 5, 1]
+            ```
+            - `legacy=False`:
+            ```python
+            >>> from transformers import T5Tokenizer
+
+            >>> tokenizer = T5Tokenizer.from_pretrained("t5-base", legacy=False)
+            >>> tokenizer.encode("Hello <extra_id_0>.")  # the extra space `[3]` is no longer here
+            [8774, 32099, 5, 1]
+            ```
+            Checkout the pull request and the issue [here](https://github.com/huggingface/transformers/pull/24565) for
+            more details.
+
+
+    Attributes:
+        sp_model (`SentencePieceProcessor`):
+            The *SentencePiece* processor that is used for every conversion (string, tokens and IDs).
+    """
+
+    vocab_files_names = VOCAB_FILES_NAMES
+    pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
+    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    model_input_names = ["input_ids", "attention_mask"]
+
+    def __init__(
+        self,
+        vocab_file,
+        eos_token="</s>",
+        unk_token="<unk>",
+        sep_token="</s>",
+        pad_token="<pad>",
+        sep_token_box=[1000, 1000, 1000, 1000],
+        pad_token_box=[0, 0, 0, 0],
+        pad_token_label=-100,
+        only_label_first_subword=True,
+        extra_ids=100,
+        loc_extra_ids=501,
+        other_extra_ids=200,
+        additional_special_tokens=None,
+        sp_model_kwargs: Optional[Dict[str, Any]] = None,
+        legacy=None,
+        **kwargs,
+    ) -> None:
+        # Add extra_ids to the special token list
+        if extra_ids > 0 and (additional_special_tokens is None or "<extra_id_0>" not in additional_special_tokens):
+            if additional_special_tokens is None:
+                additional_special_tokens = ["<extra_id_{}>".format(i) for i in range(extra_ids - 1, -1, -1)]
+            additional_special_tokens.extend(["<extra_l_id_{}>".format(i) for i in range(extra_ids - 1, -1, -1)])
+            additional_special_tokens.extend(["</extra_l_id_{}>".format(i) for i in range(extra_ids - 1, -1, -1)])
+            additional_special_tokens.extend(["<extra_t_id_{}>".format(i) for i in range(extra_ids - 1, -1, -1)])
+            additional_special_tokens.extend(["</extra_t_id_{}>".format(i) for i in range(extra_ids - 1, -1, -1)])
+
+        if loc_extra_ids > 0 and (additional_special_tokens is None or "<loc_0>" not in additional_special_tokens):
+            if additional_special_tokens is None:
+                additional_special_tokens = []
+            additional_special_tokens.extend(["<loc_{}>".format(i) for i in range(loc_extra_ids - 1, -1, -1)])
+
+        if other_extra_ids > 0 and (additional_special_tokens is None or "<other_0>" not in additional_special_tokens):
+            if additional_special_tokens is None:
+                additional_special_tokens = []
+            additional_special_tokens.extend(["<other_{}>".format(i) for i in range(other_extra_ids - 1, -1, -1)])
```

Comment:
The failing test is related to the order the tokens are added. Fast is deterministic, while slow will add additional special tokens at the end. When a new eos is passed it's added first. Fixing this can be achieved by 2 things: 1 is enforcing the added_tokens_decoder. 2 is not putting the special tokens in the convert slow to fast. (which I IMO the best solution)